### DISCUSSION TITLE
[noob question] Parsing offset and size based file

### DISCUSSION THREAD

**Discussion by nenikitov** [jam1garner/binrw#293] [Q&A]
Hello! Thank you for the crate.

I'm confused about how can I implement this file type:

```rs
#[binrw]
pub struct VfsEntry {
    offset: u32,
    size: u32,
}

#[binrw]
pub struct Vfs {
    #[br(temp)]
    #[bw(try_calc(u32::try_from(_entries.len())))]
    _file_count: u32,

    #[br(temp, count = _file_count)]
    #[bw(/* Not sure here */)]
    _entries: Vec<VfsEntry>,

    // Not sure here
    // entries: Vec<Vec<u8>>
}
```

There are these 2 examples where the header contains:
- [Sizes](https://docs.rs/binrw/latest/binrw/helpers/fn.args_iter.html#examples)
- [Offsets](https://docs.rs/binrw/latest/binrw/file_ptr/fn.parse_from_iter.html#examples)

But I'm having trouble combining these examples.

How can I create this `entries` field that contains `_file_count` entries with the corresponding slice they are pointing to?
Also, how can I generate `Vec<VfsEntry>` field when writing this file format to binary?

Thank you so much!

**Comment by latot** [comment]
I was also looking at this! hope someone can answer this :)
